File: CustomerApp/app/src/main/java/com/example/myapplication/ui/fragment/menu/MenuAdapter.java
package com.example.myapplication.ui.fragment.menu;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.fragment.app.FragmentManager;

import com.example.myapplication.databinding.CardviewFoodBinding;
import com.example.myapplication.model.Product;
import com.example.myapplication.util.android.base.BaseAdapter;

import java.util.List;

public class MenuAdapter extends BaseAdapter<MenuViewHolder, Product> {
    private final FragmentManager fragmentManager;
    private List<String> s;

    public MenuAdapter(final List<Product> items, final FragmentManager fragmentManager) {
        super(items);
        this.fragmentManager = fragmentManager;
    }

    @NonNull
    @Override
    public MenuViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new MenuViewHolder(
                CardviewFoodBinding.inflate(
                        LayoutInflater.from(parent.getContext()),
                        parent,
                        false
                ),
                fragmentManager
        );
    }
}

File: CustomerApp/app/src/main/java/com/example/myapplication/ui/fragment/menu/MenuAdapter.kt
package com.example.myapplication.ui.fragment.menu

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.core.os.bundleOf
import androidx.navigation.NavController
import com.example.myapplication.R
import com.example.myapplication.databinding.CardFoodBinding
import com.example.myapplication.model.Product
import com.example.myapplication.util.android.base.BaseAdapter
import com.example.myapplication.util.common.EXTRA_PRODUCT
import kotlinx.coroutines.ExperimentalCoroutinesApi

@ExperimentalCoroutinesApi
class MenuAdapter(
    items: MutableList<Product>,
    private val navController: NavController
): BaseAdapter<MenuViewHolder, Product>(items, true) {
    override fun getItemId(position: Int): Long = items[position].hashCode().toLong()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MenuViewHolder =
        MenuViewHolder(
            CardFoodBinding.inflate(
                LayoutInflater.from(parent.context),
                parent,
                false
            )
        ).apply {
            setOnClickListener {
                navController.navigate(
                    R.id.navigate_to_item_detail,
                    bundleOf(EXTRA_PRODUCT to items[adapterPosition])
                )
            }
        }
}